ospfd/ospf_vty.c

```diff
@@ -4884,14 +4884,37 @@ static void show_ip_ospf_nbr_nbma_detail_sub(struct vty *vty,
 static void show_ip_ospf_neighbor_detail_sub(struct vty *vty,
 					     struct ospf_interface *oi,
 					     struct ospf_neighbor *nbr,
+					     struct ospf_neighbor *prev_nbr,
 					     json_object *json, bool use_json)
 {
 	char timebuf[OSPF_TIME_DUMP_SIZE];
-	json_object *json_sub = NULL;
+	json_object *json_neigh = NULL, *json_neigh_array = NULL;
+	char neigh_str[INET_ADDRSTRLEN];
 
-	if (use_json)
-		json_sub = json_object_new_object();
-	else {
+	if (use_json) {
+		if (prev_nbr &&
+		    !IPV4_ADDR_SAME(&prev_nbr->src, &nbr->src)) {
+			json_neigh_array = NULL;
+		}
+
+		memset(neigh_str, 0, INET_ADDRSTRLEN);
```

Comment:
strlcpy handles Nul termination in dst but retains uninitialized values in fixed size array (dst). In fixed size array (dst) memset will help display only the nul terminated value in case of gdb to displays field.